Document/0x05i-Testing-Code-Quality-and-Build-Settings.md

```diff
@@ -268,23 +255,67 @@ Add the following to build.gradle:
 ### Testing for Debugging Code and Verbose Error Logging
 
 #### Overview
+StrictMode is a developer tool able to detect policy violation, e.g. disk or network access.
+It could be implemented in order to check the use of good coding practices such as no slow code or network access on the main thread.
+The policy are defined together with rules and different methods of showing the violation of a policy.
+
+There are two category of policies:
+* `StrictMode.ThreadPolicy`
+* `StrictMode.VmPolicy`
+
+The ThreadPolicy can monitor:
+* Disk Reads
+* Disk Writes
+* Network access
+* Custom Slow Code
+
+The VM policies,  applied to all threads in the virtual machine's process, are:
+* Leaked Activity objects
+* Leaked SQLite objects
+* Leaked Closable objects
+
+In order to enable `StrictMode`, is better to put the code as soon as possible, as for example in the onCreate().
+Here an example of enabling both the two kind of policies:
+```
+public void onCreate() {
+     if (DEVELOPER_MODE) {
+         StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder()
+                 .detectDiskReads()
+                 .detectDiskWrites()
+                 .detectNetwork()   // or .detectAll() for all detectable problems
+                 .penaltyLog()
+                 .build());
+         StrictMode.setVmPolicy(new StrictMode.VmPolicy.Builder()
+                 .detectLeakedSqlLiteObjects()
+                 .detectLeakedClosableObjects()
+                 .penaltyLog()
+                 .penaltyDeath()
+                 .build());
+     }
+     super.onCreate();
+ }
 
-`StrictMode` - https://developer.android.com/reference/android/os/StrictMode.html
--- TODO [Give an overview about the functionality and it's potential weaknesses] --
+```
 
+In the a release build `StrictMode` has to be disabled.
 #### Static Analysis
-
--- TODO [Add content on white-box testing for "Testing for Debugging Code and Verbose Error Logging"] --
-
+With the purpose to check if `StrictMode` is enabled you could look for the methods `StrictMode.setThreadPolicy` or `StrictMode.setVmPolicy` with high probability in the onCreate() method.
```

Comment:
Change to "`StrictMode.setVmPolicy`. Most likely they will be in the onCreate() method."